connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -481,7 +483,23 @@ public void connect() {
         upstream.sendPacket(entityPacket);
 
         CreativeContentPacket creativePacket = new CreativeContentPacket();
-        creativePacket.setContents(ItemRegistry.CREATIVE_ITEMS);
+        if (upstream.getSession().getPacketCodec().getProtocolVersion() < Bedrock_v431.V431_CODEC.getProtocolVersion()) {
+            // Pre-1.16.220 relies on item damage values that the creative content packet drops
+            ItemData[] creativeContents = new ItemData[ItemRegistry.CREATIVE_ITEMS.length];
+            for (int i = 0; i < ItemRegistry.CREATIVE_ITEMS.length; i++) {
+                ItemData item = ItemRegistry.CREATIVE_ITEMS[i];
+                if (item.getBlockRuntimeId() != 0) {
+                    creativeContents[i] = item.toBuilder().damage(ItemRegistry.getItem(item).getBedrockData()).build();
+                } else {
+                    // No block runtime ID means that this item is backwards-compatible
+                    creativeContents[i] = item;
+                }
+            }
+            creativePacket.setContents(creativeContents);
```

Comment:
This should be precomputed as it deals with constant values - no reason to run this every time a new pre 1.16.220 client joins.